src/store/transactions/transfer/transfer.thunks.ts

```diff
@@ -0,0 +1,184 @@
+
+    if (wallet !== undefined) {
+      getPoolTransactions(undefined, wallet.publicKeyCompressedHex)
+        .then((transactions) =>
+          dispatch(transferActions.loadPooledTransactionsSuccess(transactions))
+        )
+        .catch((err) => dispatch(transferActions.loadPooledTransactionsFailure(err)));
+    }
+  };
+}
+
+/**
+ * Fetches the accounts to use in the transaction. If the transaction is a deposit it will
+ * look for them on Ethereum, otherwise it will look for them on the rollup api
+ */
+function fetchAccounts(
+  fromItem: number | undefined,
+  pooledTransactions: PooledTransaction[],
+  fiatExchangeRates: FiatExchangeRates,
+  preferredCurrency: string
+): AppThunk {
+  return (dispatch: AppDispatch, getState: () => AppState) => {
+    const {
+      global: { wallet, tokensPriceTask },
+    } = getState();
+
+    if (wallet !== undefined) {
+      dispatch(transferActions.loadAccounts());
+
+      return CoordinatorAPI.getAccounts(wallet.publicKeyBase64, undefined, fromItem)
+        .then((res) => {
+          const accounts = res.accounts.map((account) =>
+            createAccount(
+              account,
+              pooledTransactions,
+              undefined,
+              tokensPriceTask,
+              fiatExchangeRates,
+              preferredCurrency
+            )
+          );
+
+          return { ...res, accounts };
+        })
+        .then((res) => dispatch(transferActions.loadAccountsSuccess(res)))
+        .catch((err) => dispatch(transferActions.loadAccountsFailure(err)));
+    }
+  };
+}
+
+/**
+ * Fetches the recommended fees from the Coordinator
+ * @returns {void}
+ */
+function fetchFees(): AppThunk {
+  return function (dispatch: AppDispatch, getState: () => AppState) {
+    const {
+      global: { coordinatorStateTask },
+    } = getState();
+
+    if (
+      coordinatorStateTask.status === "successful" ||
+      coordinatorStateTask.status === "reloading"
+    ) {
+      const nextForger = getNextBestForger(coordinatorStateTask.data);
+
+      if (nextForger !== undefined) {
+        dispatch(transferActions.loadFees());
+
+        return CoordinatorAPI.getState({}, nextForger.coordinator.URL)
+          .then((res) => dispatch(transferActions.loadFeesSuccess(res.recommendedFee)))
+          .catch((err) => dispatch(transferActions.loadFeesFailure(err)));
+      }
+    }
+  };
+}
+
+function transfer(amount: BigNumber, from: Account, to: Partial<Account>, fee: number): AppThunk {
+  return (dispatch: AppDispatch, getState: () => AppState) => {
+    const {
+      global: { wallet, coordinatorStateTask },
+    } = getState();
+
+    if (
+      wallet !== undefined &&
+      (coordinatorStateTask.status === "successful" || coordinatorStateTask.status === "reloading")
+    ) {
+      dispatch(transferActions.startTransactionApproval());
+
+      const nextForgerUrls = getNextForgerUrls(coordinatorStateTask.data);
+      const txData = {
+        type: "Transfer" as const,
```

Comment:
This `as const` is kinda weird also, this is because of the enum setup in the declarations file right?